onedal/covariance/incremental_covariance.py

```diff
@@ -58,10 +58,48 @@ def __init__(self, method="dense", bias=False, assume_centered=False):
         self._reset()
 
     def _reset(self):
+        self._need_to_finalize = False
         self._partial_result = self._get_backend(
             "covariance", None, "partial_compute_result"
         )
 
+    def __getstate__(self):
+        self.finalize_fit()
+        data = self.__dict__.copy()
+        partial_result_data = dict()
+        partial_result_data["partial_n_rows"] = from_table(
+            data["_partial_result"].partial_n_rows
+        )
+        partial_result_data["partial_crossproduct"] = from_table(
+            data["_partial_result"].partial_crossproduct
+        )
+        partial_result_data["partial_sums"] = from_table(
+            data["_partial_result"].partial_sums
+        )
+        data["_partial_result"] = partial_result_data
+        data.pop("_policy", None)
+
+        return data
+
+    def __setstate__(self, data):
+        partial_result = self._get_backend("covariance", None, "partial_compute_result")
+        if data["_partial_result"]["partial_n_rows"].size > 0:
+            partial_result.partial_n_rows = to_table(
+                data["_partial_result"]["partial_n_rows"]
+            )
+        if data["_partial_result"]["partial_crossproduct"].size > 0:
+            partial_result.partial_crossproduct = to_table(
+                data["_partial_result"]["partial_crossproduct"]
+            )
+        if data["_partial_result"]["partial_sums"].size > 0:
+            partial_result.partial_sums = to_table(
+                data["_partial_result"]["partial_sums"]
+            )
+
+        data["_partial_result"] = partial_result
```

Comment:
I worry about checking against .size. Is it always initialized to 0? Or is hasattr a better solution? Not sure.

Reply:
could you clarify please what is wrong with size checking? if the tables from getstate are empty, they are converted to empty list (not sure it is good practice but anyway) and here I try to check if getstate returned empty list to avoid providing it into to_table.